src/shared/ui/components/BaseTable.vue

```diff
@@ -125,9 +126,23 @@ const segmentInfo = computed(() => {
   if (!props.pagination) return '';
 
   const p = props.pagination;
-  const start = (p.page - 1) * p.page_size + 1;
-  const end = p.page * p.page_size;
-  return `${start}—${end > p.total ? p.total : end} of ${p.total}`;
+
+  if (!p.total_items) p.total_items = props.items.length;
+
+  if (props.sticky) {
+    if (p.per_page > props.items.length) return `${props.items.length}—1 of ${p.total_items}`;
+
+    const start = (p.page - 1) * p.per_page + props.items.length;
+
+    const end = start - props.items.length + 1;
+
+    return `${start}—${end} of ${p.total_items}`;
+  }
+
+  const start = (p.page - 1) * p.per_page + 1;
+  const end = p.page * p.per_page;
+
+  return `${start}—${end > p.total_items ? p.total_items : end} of ${p.total_items}`;
```

Comment:
This functionality seems a bit complicated. It would help to separate the complicated logic into some external function and cover it with unit tests thoroughly.